packages/legend-application-studio/src/components/editor/edit-panel/GrammarTextEditor.tsx

```diff
@@ -811,10 +812,14 @@ export const GrammarTextEditor = observer(() => {
       });
       _editor.onDidChangeModelContent(() => {
         grammarTextEditorState.setGraphGrammarText(getEditorValue(_editor));
-        editorStore.graphState.clearCompilationError();
         // we can technically can reset the current element label regex string here
         // but if we do that on first load, the cursor will not jump to the current element
         // also, it's better to place that logic in an effect that watches for the regex string
+
+        editorStore.graphState.clearCompilationError();
+
+        clearProblemMarkers();
+        editorStore.graphState.setIsStaleTextWarnings(true);
```

Comment:
I think we talked about this, this should actually just clear the warnings, there's not even need to call `setIsStale...` at that point